.devcontainer/README.md

```diff
@@ -0,0 +1,236 @@
+
+**Solution**:
+```bash
+# Run the quick setup script
+bash .devcontainer/quick-setup.sh
+
+# Or manually build in dependency order:
+cd packages/tsconfig && pnpm build  # Config packages first
+cd packages/configraph && npx pnpm build  # Use npx for workspace deps
+cd packages/core && npx pnpm build
+```
+
+**Why this works**: Using `npx pnpm build` ensures the build happens in the workspace context where workspace dependencies are properly resolved.
+
+#### 2. `pnpm install` hangs or fails
+
+**Cause**: Network issues, registry problems, or interactive prompts in non-interactive environment.
+
+**Solution**:
+```bash
+# Set non-interactive environment
+export CI=true
+
+# Try alternative install methods
+pnpm install --ignore-scripts --no-optional
+# or
+pnpm install --frozen-lockfile=false --force
+```
+
+#### 3. Global tools not found (`turbo`, `tsup`, etc.)
+
+**Cause**: Tools weren't installed in the onCreate phase or installation failed.
+
+**Solution**:
+```bash
+# Install missing tools
+npm install -g turbo@latest tsup@latest typescript@latest
+
+# Verify installation
+turbo --version
+tsup --version
+tsc --version
+```
+
+#### 4. Build fails with "Local package.json exists, but node_modules missing"
+
+**Cause**: Package-level node_modules weren't installed properly.
+
+**Solution**:
+```bash
+# Force reinstall from workspace root
+pnpm install --force
+
+# Or reinstall specific package
+cd packages/failing-package
+pnpm install
+```
+
+#### 5. Workspace packages can't find each other
+
+**Cause**: Workspace linking isn't working properly.
+
+**Solution**:
+```bash
+# Verify workspace structure
+pnpm list --depth=0
+
+# Check if packages are properly linked
+ls -la packages/tsconfig/  # Should show config files
+ls -la packages/configraph/node_modules/@dmno/  # Should show linked packages
+```
+
+## Package Dependencies
+
+Understanding the dependency order helps with troubleshooting:
+
+### Configuration Packages (no dependencies)
+- `@dmno/tsconfig` - TypeScript configurations
+- `@dmno/eslint-config` - ESLint configurations  
+
+### Core Libraries (minimal dependencies)
+- `@dmno/encryption-lib` - Encryption utilities
+- `@dmno/ts-lib` - TypeScript utilities
+- `@dmno/ui-lib` - UI components
+
+### Engine Packages (depend on config + core)
+- `@dmno/configraph` - Configuration graph engine
+- `@dmno/core` - Main DMNO engine
+
+### Integrations (depend on core)
+- `@dmno/vite-integration`
+- `@dmno/astro-integration`
+- `@dmno/nextjs-integration`
+- `@dmno/remix-integration`
+- `@dmno/fastify-integration`
+
+### Plugins (depend on core)
+- `@dmno/encrypted-vault-plugin`
+- `@dmno/1password-plugin`
+- `@dmno/bitwarden-plugin`
+- `@dmno/infisical-plugin`
+
+### Platforms (depend on core)
+- `@dmno/netlify-platform`
+- `@dmno/vercel-platform`
+- `@dmno/cloudflare-platform`
+
+**Build Rule**: Always build packages in dependency order. Configuration packages first, then core libraries, then everything else.
```

Comment:
Hopefully folks wont have to think about this too much. A single run of `pnpm run build:libs` uses turborepo to build everything in the correct dependency order, and then you can work on individual packages.